src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/PostActionModel.cs

```diff
@@ -30,93 +32,105 @@ internal class PostActionModel : ConditionedConfigurationElementBase, IPostActio
                     }
 
-                    if (actionLocalizations?.Instructions.Count > localizedManualInstructions)
+                    if (actionLocalizations?.Instructions.Count > localizedInstructionCount)
                     {
                         // Localizations provide more translations than the number of manual instructions we have.
-                        string excessInstructionLocalizationIndexes = string.Join(
+                        string excessInstructionLocalizationIds = string.Join(
                             ", ",
-                            actionLocalizations.Instructions.Keys.Where(k => k < 0 || k > instructionOptions.Count).Select(k => k.ToString()));
-                        host.Logger.LogWarning(string.Format(LocalizableStrings.Authoring_InvalidManualInstructionLocalizationIndex, excessInstructionLocalizationIndexes, postActionIndex));
+                            actionLocalizations.Instructions.Keys.Where(k => !localizedInstructions.Any(i => i.Id == k)));
+                        host.Logger.LogWarning(string.Format(LocalizableStrings.Authoring_InvalidManualInstructionLocalizationIndex, excessInstructionLocalizationIds, postActionId));
                     }
                 }
 
                 PostActionModel model = new PostActionModel(
-                    actionLocalizations?.Description ?? action.ToString(nameof(model.Description)),
+                    postActionId,
+                    actionLocalizations?.Description ?? action.ToString(nameof(Description)),
                     action.ToGuid(nameof(ActionId)),
-                    action.ToBool(nameof(model.ContinueOnError)),
+                    action.ToBool(nameof(ContinueOnError)),
                     args,
-                    instructionOptions,
-                    action.ToString(nameof(model.Condition))
+                    localizedInstructions,
+                    action.ToString(nameof(Condition))
                 );
 
-                modelList.Add(model);
+                localizedPostActions.Add(model);
             }
 
-            if (localizations?.Count > localizedPostActions)
+            if (localizations?.Count > localizedPostActionCount)
             {
                 // Localizations provide more translations than the number of post actions we have.
-                string excessPostActionLocalizationIndexes = string.Join(", ", localizations.Keys.Where(k => k < 0 || k > modelList.Count).Select(k => k.ToString()));
-                host.Logger.LogWarning(string.Format(LocalizableStrings.Authoring_InvalidPostActionLocalizationIndex, excessPostActionLocalizationIndexes));
+                string excessPostActionLocalizationIds = string.Join(", ", localizations.Keys.Where(k => !localizedPostActions.Any(p => p.Id == k)).Select(k => k.ToString()));
+                host.Logger.LogWarning(string.Format(LocalizableStrings.Authoring_InvalidPostActionLocalizationIndex, excessPostActionLocalizationIds));
```

Comment:
do we have integration tests for these warnings? If not, it would be good to add ones.

Reply:
We have that planned with https://github.com/dotnet/templating/issues/3132